src/utils.jl

```diff
@@ -109,12 +109,15 @@ not the same as choosing them uniform in cartesian coordinates!).
 If `r` is unitless, then the unit of the system is applied. 
 """
 function rattle!(at::FlexibleSystem, r::Quantity)
+   (length(at.particles) > 0) || return at
+   at_unit = unit(position(at, 1)[1])
+   r = uconvert(at_unit, r)
    for i = 1:length(at.particles)
       p = at.particles[i]
       𝐫ᵢ = p.position 
       T = typeof(ustrip(𝐫ᵢ[1]))
       ui = randn(Vec3{T})
-      p_new = _set_position(p, 𝐫ᵢ + r * ui / norm(ui))
+      p_new = _set_position(p, 𝐫ᵢ + rand() * r * ui / norm(ui))
```

Comment:
This new `rand()` is for consistency with the docstring, to have uniform sampling in the radial component (without this we are sampling the surface of the sphere of radius exactly r)